db/migrate/20151017171416_add_rpush.rb

```diff
@@ -0,0 +1,349 @@
+      change_column :rapns_apps, :environment, :string, null: false
+      change_column :rapns_apps, :certificate, :text, null: false
+
+      change_column :rapns_notifications, :error_description, :string, null: true, default: nil
+      change_column :rapns_notifications, :sound, :string, default: '1.aiff'
+
+      rename_column :rapns_notifications, :data, :attributes_for_device
+      rename_column :rapns_apps, :name, :key
+
+      remove_column :rapns_apps, :auth_key
+
+      remove_column :rapns_notifications, :collapse_key
+      remove_column :rapns_notifications, :delay_while_idle
+      remove_column :rapns_notifications, :registration_ids
+      remove_column :rapns_notifications, :retries
+
+      add_column :rapns_notifications, :app, :string, null: true
+
+      AddGcm::Rapns::Notification.reset_column_information
+      AddGcm::Rapns::App.reset_column_information
+
+      AddGcm::Rapns::App.all.each do |app|
+        AddGcm::Rapns::Notification.where(app_id: app.id).update_all(app: app.key)
+      end
+
+      if index_name_exists?(:rapns_notifications, :index_rapns_notifications_multi, true)
+          remove_index :rapns_notifications, name: :index_rapns_notifications_multi
+      end
+
+      remove_column :rapns_notifications, :app_id
+
+      add_index :rapns_notifications, [:delivered, :failed, :deliver_after], name: :index_rapns_notifications_multi
+    end
+  end
+
+  class AddWpns < ActiveRecord::Migration
+    module Rapns
+      class Notification < ActiveRecord::Base
+        self.table_name = 'rapns_notifications'
+      end
+    end
+
+    def self.up
+      add_column :rapns_notifications, :uri, :string, null: true
+    end
+
+    def self.down
+      AddWpns::Rapns::Notification.where(type: 'Rapns::Wpns::Notification').delete_all
+      remove_column :rapns_notifications, :uri
+    end
+  end
+
+  class AddAdm < ActiveRecord::Migration
+    module Rapns
+      class Notification < ActiveRecord::Base
+        self.table_name = 'rapns_notifications'
+      end
+    end
+
+    def self.up
+      add_column :rapns_apps, :client_id, :string, null: true
+      add_column :rapns_apps, :client_secret, :string, null: true
+      add_column :rapns_apps, :access_token, :string, null: true
+      add_column :rapns_apps, :access_token_expiration, :datetime, null: true
+    end
+
+    def self.down
+      AddAdm::Rapns::Notification.where(type: 'Rapns::Adm::Notification').delete_all
+
+      remove_column :rapns_apps, :client_id
+      remove_column :rapns_apps, :client_secret
+      remove_column :rapns_apps, :access_token
+      remove_column :rapns_apps, :access_token_expiration
+    end
+  end
+
+  class RenameRapnsToRpush < ActiveRecord::Migration
+    module Rpush
+      class App < ActiveRecord::Base
+        self.table_name = 'rpush_apps'
+      end
+
+      class Notification < ActiveRecord::Base
+        self.table_name = 'rpush_notifications'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.